### ISSUE TITLE
Create a spreadsheet for rule building for Individual Measures

### ISSUE THREAD

**Issue by meggerc** [aethelwulffe/PQRS_Gateway#11]
A text file is attached that includes the ICD, CPT and HCPCS codes used for the denominator of each measure in the Diabetes Mellitus (DM) Measure Group. This measure group includes 6 individual measures. 

The first field in each row states the PQRS number and type of code, e.g., PQRS###_ICD9.

I used CPT as the row header for both CPT and HCPCP codes; some measures include CPT, other HCPCS, and others include both CPT AND HCPCS. 

A common denominator format is ICD9 (for 1/1/15 - 9/30/15, ICD10 (for 10/1/15-12/31/15 AND CPT.

Some measures have only CPT codes; others (not in this group) will have a lot of this and this OR this and that, without, etc.... the DM measure group is pretty straightforward.

Age rules:
PQRS1 18-75
PQRS110 6 months and older
PQRS117 18-75
PQRS119 18-75
PQRS163 18-75
PQRS226 18 and older

[PQRS DM Measure Group.txt](https://github.com/aethelwulffe/PQRS_Gateway/files/127189/PQRS.DM.Measure.Group.txt)

![image](https://cloud.githubusercontent.com/assets/16786878/12991657/1d740028-d0de-11e5-9d29-f9a1e92a3461.png)

**Comment by meggerc** [comment]
Some individual measures will belong to more than one measure group.

**Comment by meggerc** [comment]
Individual measures that belong to more than one measure group. 

![image](https://cloud.githubusercontent.com/assets/16786878/12995034/56084c0a-d0f2-11e5-8310-d5f7f39a8a37.png)

**Comment by aethelwulffe** [comment]
Our code types within Openemr are:

**Comment by aethelwulffe** [comment]
Sorry...I accidentally closed this issue!

**Comment by aethelwulffe** [comment]
Meg,
  We are totally ready to code measures.
You got the data,
I got some sample queries,
Brian has some skeleton code,
Sam has got some important bits mapped out,
We have a patient data import tool.
We are go for launch.

When do you think we can get together again to write a couple of measures and start the code factory?

**Comment by meggerc** [comment]
My new schedule is Monday-Friday 1130-8. I'm free this weekend. Are you?

**Comment by meggerc** [comment]
And that's awesome news!

**Comment by aethelwulffe** [comment]
I'll have to be!

**Comment by meggerc** [comment]
I'm sorry Art. I'm frazzled so I'm finishing up at home while I tend to
other things like laundry and making appointments...you can pick any time
Saturday or Sunday.

**Comment by aethelwulffe** [comment]
No sweat mon!
Glad to get you any day, any time.
Just finished up the draft of PQRS_0001!  Untested and full of ugly 
comments!
With the skeleton templates, some complete examples and your collated 
data, we can bust these out like a shashimi chef chopping onions!

**Comment by aethelwulffe** [comment]
Lets post file revisions to this issue again.
Hopefully we will be back on track soon.  Lost some serious time due to my personal weaknesses.
I will be posting the pseudocode version of Measure 21 on assumptions about the database row label columns and the table name.  I can convert it once we have the actual table list.

**Comment by meggerc** [comment]
@aethelwulffe - here is the 2 column table for the community/pop health domain.

 [POPH_Domain_Code_Table_2_column.txt](https://github.com/aethelwulffe/PQRS_Gateway/files/145109/POPH_Domain_Code_Table_2_column.txt)

**Comment by meggerc** [comment]
Communication and Care Coordination (CCCO)
Measures that have more than one 'column' that is b or greater after the measure # follow these rules:
PQRS_0185 =  (A + NOT am)
PQRS_0217-0220 = ((a + (b or e) OR (c + (d or e))
PQRS_0320 = (a + NOT am)
PQRS_0325 = Major Depressive Disorder (MDD) = a, chronic condition = b-f
PQRS_0325 = ((a + b) OR (a + c) OR (a + d) OR (a + e) OR (a + f)) 

[CCCO_Domain_Code_Table_2_column.txt](https://github.com/aethelwulffe/PQRS_Gateway/files/145387/CCCO_Domain_Code_Table_2_column.txt)

**Comment by meggerc** [comment]
Efficiency and Cost Reduction (ECR)

PQRS_0224 (a OR b)

[ECR_Domain_Code_Table_2_column.txt](https://github.com/aethelwulffe/PQRS_Gateway/files/145508/ECR_Domain_Code_Table_2_column.txt)

**Comment by aethelwulffe** [comment]
Geez...wish I could dig into these immediately.  I am coding blind at 
the moment!  We have been on every other domain but the ones you have 
done!  I should stop with the measures and start making tables in all 
honesty.

**Comment by meggerc** [comment]
I'll have 2 more by 8 tonight and hopefully the biggest one - the Effective Clinical Care by midnight. I'm proofing and reviewing so mistakes don't catch up with us later.

**Comment by meggerc** [comment]
2 more domains I mean

**Comment by meggerc** [comment]
Person-Centered Experience and Outcomes (PTCT)
PQRS_0143 = ((a + b) OR (a + C))
PQRS_0144 IF PQRS_0143 numerator = 1125F
PQRS_0144 = ((a + b) OR (a + C))
PQRS_0303 = (a + NOT am)
PQRS_0304 = (a + NOT am)
PQRS_0398 = (a + NOT b)

[PTCT_Domain_Code_Table_2_column.txt](https://github.com/aethelwulffe/PQRS_Gateway/files/145726/PTCT_Domain_Code_Table_2_column.txt)

**Comment by aethelwulffe** [comment]
Cool megger!
I am on my way to the halfway point on the measures!  Since I just 
believe in the existence of the mythical tables of meggerdom, it is 
REALLY easy to code the measures!
Just did measure 109, so I am not up to these here yet.  With Brian 
needing to make sure our database server
(production) is configured the best it can be for the purpose, I could 
use your help going back over the code and checking for logic and naming 
(and syntax of course).  When I needed a table name I would do something 
like 'pqrs_patient_safety" and for this one I would have chosen 
'pqrs_pceo'...which you are calling PTCT...which means that you had a 
reason for that from some official list probably, and we will need to 
change my code.

**Comment by meggerc** [comment]
@aethelwulffe 
Actually PTCT is person centered, which used to be Patient (PT) centered (CT). I listed the acronyms in Issue #12. Here they are again:
![image](https://cloud.githubusercontent.com/assets/16786878/13330038/0aa3f4e8-dbc5-11e5-87af-1535e99477f7.png)
I'll be glad to go oer the code and check for logic. Can we get together on Saturday again?

**Comment by leebc** [comment]
@meggerc Is "pqrs_0185am" the correct nomenclature for a couple of these?

**Comment by meggerc** [comment]
Yes, it means it's a modifier for pqrs_0185a, per Art's request

**Comment by leebc** [comment]
Excellent.  I created/committed SQl commands to insert all this data into our database.  The "am"'s were  outliers from the rest of the data, so I wanted to make sure it was correct.
@meggerc How would you like me to attribute your contributions of authorship?  As "meggerc" or your full name or something else/additional?

**Comment by leebc** [comment]
(Reporting for Art)   Domain CCCO: Measure 325 needs to be reviewed.  Different code sets are bending into each other.    Set "325a" should have 20 values; "325b should have 15; "325c" should be everything else.

**Comment by meggerc** [comment]
@leebc  - how do you attribute your contributions? this is all new to me

**Comment by leebc** [comment]
I've been using "leebc", Art uses his full name.  I put in meggerc for you, but can change fairy easiy.

**Comment by meggerc** [comment]
OK. meggerc is good enough. :) thanks

**Comment by aethelwulffe** [comment]
I actually have not referenced modifiers (yet).  I have been hard-coding those...only a few instances.  All numerators are hard coded as well.  All the same, it is good to have standards to do all of that.  False starts or otherwise, the end result is that as a team we have been each equally instrumental in getting a draft of the measures code stood up, and a plan for tweaking them in an automated fashion.  At this second, I have four more measures to code the first draft of.  We WILL be testing before March.  That is unbelievable.

**Comment by meggerc** [comment]
Patient Safety (PTSF)

PQRS_0156 = (a + NOT b)
PQRS_0192 = (a + NOT am + NOT b)

PQRS_0258 
if male:    = (a + NOT bm)
if female: = (a + NOT bf) 

PQRS_0259 
if male:    = (a + NOT bm)
if female: = (a + NOT bf) 

PQRS_0260 = (a + NOT b)
PQRS_0347
if male:    = (a + NOT bm)
if female: = (a + NOT bf) 

PQRS_0348 = (a + b + NOT am)
PQRS_0348.1 = (a + b + NOT am)
PQRS_0383 =( (a+b) + c OR d OR e OR f OR g) + NOT h)) 
PQRS_0388 =  (a + NOT am)


Other measures with two reporting options
PQRS_0238 AND PQRS_0238.1 
PQRS_0393 AND PQRS_0393.1

[PTSF_Domain_Code_Table_2_column.txt](https://github.com/aethelwulffe/PQRS_Gateway/files/148867/PTSF_Domain_Code_Table_2_column.txt)

**Comment by meggerc** [comment]
@aethelwulffe Have you loaded the code tables yet? I'm looking over the text files I created to make sure there I was consistent with the measure numbers for the measures with more than one option. I'm also checking for dropped trailing zeros. Once I'm done proofing I can post them to GitHub.

**Comment by meggerc** [comment]
@aethelwulffe Should I post one CSV file with all the codes instead of by domain? I split it up by domain to make it more manageable on my end.

**Comment by meggerc** [comment]
Here are the revised and proofed tables for all domains except for effective clinical care. I'll post that one later today.
[Two_column_tables.zip](https://github.com/aethelwulffe/PQRS_Gateway/files/151141/Two_column_tables.zip)

**Comment by aethelwulffe** [comment]
We have not actually (as far as I know) loaded the table into a database 
yet.  Bryan made the query to do so, and set up his magic process so 
that we can do it again and again as corrections come in.  Perfect or 
not, we will be loading one up very soon for testing, perhaps today.
   Keep things manageable on your end.  after all, we are using a 
different table per domain to make it manageable on OUR end.

**Comment by meggerc** [comment]
OK! All but one table are in the zIp file I just posted.

**Comment by leebc** [comment]
Created insert_PQRS_xxxx_values.sql from files in today's .zip

Reviewed codes against PDF.  For the domains listed below, the beginnings/ends of the values match against their measure numbers.  

Domain  `pqrs_poph` includes codes for these measures: 
pqrs_0110a
pqrs_0111a
pqrs_0128a
pqrs_0131a
pqrs_0134a
pqrs_0173a
pqrs_0226a
pqrs_0317a
pqrs_0394a
pqrs_0402a

**Comment by leebc** [comment]
Domain ECR:
Broke pqrs_0065a into pqrs_0065a and pqrs_0065b.  Validated.
Broke pqrs_0066a into pqrs_0066a, pqrs_0066b, pqrs_0066c.  (C may probablyt be needed.) Validated.


* Validated means I went through the entire list of codes.

**Comment by meggerc** [comment]
Why are you breaking them up? Art thought we could keep the 'ands' in the
same column. If not, I'll do the last domain according to the way you want
it so you don't have to break them.

**Comment by leebc** [comment]
@meggerc:  It appears that the ANDs need to be different code type thingies.  You might want to check with Art on that.

If you're going to rework it, I found 2 errors in  ptct: There's one spot with a blank line (row) and one spot for measure 143 where the code is 'D37.8D37.9', but it should be 2 codes.
Also, somewhere you have "pqrs_109_a" instead of "pqrs_0109_a".  
Otherwise, everything massages well into SQL inserts.
We corrected the issues I mentioned above, so don't worry about it unless you're otherwise creating new files with updates.


@aethelwulffe : Be advised that 'pqrs_0258' and pqrs_'0259' and 'pqrs_0347' have something weird going on with an "_b_f" modifier that isn't  immediately obvious to me. 
Also, the tables go through "_h" right now.

**Comment by aethelwulffe** [comment]
Well, those measures are not actually in our completed "short list", so I have not actually reviewed those.  I can say that after messing with one for over an hour, I discovered that instead of something like 7 code groups, I discovered that it was actually only two.  The writer was simply an idiot.  That was not apparent from the use of "AND OR" statements, just from reading the text.  There is also a measure that repeats the same codes.  Straight up typo.

**Comment by meggerc** [comment]
If more than one measure have the same codes it may be that there is one additional piece to the instructions that varies betweeen the codes. There were a few measures that had to do with different injuries - back, neck, etc. The codes are the same but there was a manual piece that differentiated the measures. I listed all codes separately to be consistent. There aren't many that repeat the same codes. 

I'm not sure what you mean by the first couple of sentences. I'll call you. I think it has do with  the code type thingiess  Bryan referred to? LOL

**Comment by aethelwulffe** [comment]
Oh, we were told (by lana) that we only really needed about 80 of the measures for the first round of reporting. so we only coded those.  258, 259, and 347 are among those that we have not done yet.  That does not mean that we are not going to do them, so having the codes in the database is a good thing.

**Comment by meggerc** [comment]
@leebc I talked to Art about the code thingies. I was including the ICD and CPT in one column e.g., (a). I'm going to split them up. Do you want me to redo all the tables? it's not a problem, but if you already fixed a table I don't want to do rework.

**Comment by meggerc** [comment]
@aethelwulffe @aethelwulffe  Be advised that 'pqrs_0258' and pqrs_'0259' and 'pqrs_0347' have something weird going on with an "_b_f" modifier that isn't immediately obvious to me. 
Also, the tables go through "_h" right now.

The measure has different rules for male and female so b f is for females. Only one measure goes through h- it's for a major depressive disorder AND one chronic condition among several.

**Comment by leebc** [comment]
@aethelwulffe 
For pqrs_ccco
pqrs_0024.01_a
pqrs_0024_a
pqrs_0046_a
pqrs_0047_a
pqrs_0081_a
pqrs_0137_a
pqrs_0138.01_a
pqrs_0138_a
pqrs_0141_a
pqrs_0147_a
pqrs_0155_a
pqrs_0182_a
pqrs_0185_a
pqrs_0185_am
pqrs_0217_a
pqrs_0217_b
pqrs_0217_c
pqrs_0217.01_a
pqrs_0217.01_b
pqrs_0217.01_c
pqrs_0218_a
pqrs_0218_b
pqrs_0218_c
pqrs_0218.01_a
pqrs_0218.01_b
pqrs_0218.01_c
pqrs_0219_a
pqrs_0219_b
pqrs_0219_c
pqrs_0219.01_a
pqrs_0219.01_b
pqrs_0219.01_c
pqrs_0220_a
pqrs_0220_b
pqrs_0220_c
pqrs_0220.01_a
pqrs_0220.01_b
pqrs_0220.01_c
pqrs_0221_a
('pqrs_0221_b
('pqrs_0221_c
('pqrs_0221.01_a
('pqrs_0221.01_b
('pqrs_0221.01_c
('pqrs_0222_a
('pqrs_0222_b
('pqrs_0222_c
('pqrs_0222.01_a
('pqrs_0222.01_b
('pqrs_0222.01_c
('pqrs_0223_a
('pqrs_0223_b
('pqrs_0223_c
('pqrs_0223.01_a
('pqrs_0223.01_b
('pqrs_0223.01_c
('pqrs_0225_a
('pqrs_0261_a
('pqrs_0265_a
('pqrs_0320_a
('pqrs_0320_am
('pqrs_0325_a
('pqrs_0325_b
('pqrs_0325_c
('pqrs_0325_d
('pqrs_0325_e
('pqrs_0325_f
('pqrs_0336_a
('pqrs_0391_a
('pqrs_0395_a
('pqrs_0396_a
('pqrs_0397_a

**Comment by leebc** [comment]
@aethelwulffe ist of which measures belong to which table.  This does NOT include EFCC, which I don't have yet.
[measures_per_domain.txt](https://github.com/aethelwulffe/PQRS_Gateway/files/153655/measures_per_domain.txt)

**Comment by aethelwulffe** [comment]
Cool beans.
258 looks like using a table is overkill.  Only four codes or so to check.  I don't mind ones like that, even if I have a little more logic to check "if female else".  I have already done some of those, and the ones where it is different number of encounters are a lot harder.
Even though 258 is really short, let's keep the conditional logic when it is a F/M distinction.  Might be others with a lot more codes.

**Comment by aethelwulffe** [comment]
Poph and PTCT are verified and in the bag.

**Comment by meggerc** [comment]
@aethelwulffe I'm finishing up EFCC today

**Comment by meggerc** [comment]
Here is the final table. Effective Clinical Care (EFCC). I proofed it but it needs another set of eyes.

PQRS_0164 - PQRS_0168 give a choice of CPT OR CPT + CPT. I only included the first set of CPT codes because they are exactly the same for both choices and the first choice is less restrictive.

If a measure includes a group of codes with one or two codes, I didn't include them since it seems that you're hard-coding these. If a measure ONLY includes one or two codes total, or in one column, I included them to make sure they don't get missed.

I got rid of some duplicates in a couple of the larger measures. I caught a few codes where 11 was used instead of H, and 1 was used instead of T for the ICD10 codes.

[EFCC_Domain_Code_Table_2_column.zip](https://github.com/aethelwulffe/PQRS_Gateway/files/157330/EFCC_Domain_Code_Table_2_column.zip)
